### PR TITLE
nightly channel updates

### PR DESCRIPTION
**Pull request by duanyutong**

### CODE REVIEW

File: drake-torch-melodic.dockerfile [hunk 1]
```diff
@@ -35,7 +35,8 @@ RUN apt-get update && apt-get install -qy \
     python3-rosinstall
 # bootstrap rosdep
 RUN rosdep init && rosdep update
-ENV ROS_DISTRO melodic
+ENV ROS_DISTRO=melodic
```
Comment (aneksteind) [inline drake-torch-melodic.dockerfile:38]: we're not on 20.04 with the containers now, right?

**Review by aneksteind — APPROVED** [review]
Looks fine but I'd wait for @brook-stevens or @ankitm123

File: drake-torch-melodic.dockerfile [hunk 2]
```diff
@@ -123,6 +124,8 @@ RUN python3 -m pip install --upgrade --no-cache-dir --compile \
 
 RUN add-apt-repository -y ppa:git-core/ppa \
     && apt-get install -qy \
+        openssh-server \
```
Comment (ankitm123) [inline drake-torch-melodic.dockerfile:127]: May be not for this PR, but we should pin the versions of the packages.

File: drake-torch-noetic.dockerfile [hunk 3]
```diff
@@ -60,92 +48,74 @@ RUN apt-get update && apt-get install -qy \
...
+#     && ./b2 install threading=multi link=shared
+
+# yaml-cpp 0.6.3
+RUN curl -SL https://github.com/jbeder/yaml-cpp/archive/yaml-cpp-0.6.3.tar.gz | tar -xz \
+    && cd yaml-cpp-yaml-cpp-0.6.3 \
+    && mkdir build \
+    && cd build \
+    && cmake .. -D YAML_BUILD_SHARED_LIBS=ON \
+    && make install -j 12 \
+    && rm -rf $HOME/yaml-cpp-yaml-cpp-0.6.3
```
Comment (ankitm123) [inline drake-torch-noetic.dockerfile:72]: use double quotes.

**Comment by ankitm123** [comment]
We should start using [hadolint](https://github.com/hadolint/hadolint) to lint our dockerfiles. Ideally it should be added to the pipeline.

**Comment by ankitm123** [comment]
@duanyutong imo the hadolint changes can be done in a separate PR, good to merge for now.

**Review by ankitm123 — APPROVED** [review]

File: drake-torch-noetic.dockerfile (hunk 3 above)
Comment (duanyutong) [reply drake-torch-noetic.dockerfile:72]: done

File: drake-torch-melodic.dockerfile (hunk 2 above)
Comment (ankitm123) [reply drake-torch-melodic.dockerfile:127]: With so many packages updated (we have to look at the logs to even know which one was updated), it's difficult to track down which update broke a build (if that happens). Do we really need to update everything? If not, then just select a few packages and update them, and let the tag reflect that (or attach metadata to the image using the `LABEL` keyword): https://www.balena.io/blog/our-dockerfile-tips-tricks/. 
As for doing it in an automated way, I dont have an answer now, let me think abt it a bit more, and get back to you. But I feel if things are not broken with the version already present (or we dont need an update to a version), should we even update them?

Comment (duanyutong) [reply drake-torch-melodic.dockerfile:127]: I think we should. We've been plagued with issues from using old packages.

**Pull request merged**